rollingpin/status.py

```diff
@@ -0,0 +1,65 @@
+import json
+import logging
+import posixpath
+import urlparse
+
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks, returnValue, CancelledError
+from twisted.web.client import Agent, readBody
+
+
+TIMEOUT = 5  # seconds
+
+
+# this is the "safe" response so if we're not configured or something goes
+# wrong we'll just not prompt the user so deploys aren't blocked.
+SAFE_DEFAULT = {"time_status": "work_time", "busy": False}
+
+
+@inlineCallbacks
+def _do_status_http_request(harold_base_url):
+    base_url = urlparse.urlparse(harold_base_url)
+    path = posixpath.join(base_url.path, "harold/deploy/status")
+    url = urlparse.urlunparse((
+        base_url.scheme,
+        base_url.netloc,
+        path,
+        None,
+        None,
+        None
+    ))
+
+    agent = Agent(reactor)
+    response = yield agent.request("GET", url)
+    if response.code != 200:
+        raise Exception("harold responded with an error: %d" % response.code)
+    body = yield readBody(response)
+    returnValue(json.loads(body))
+
+
+@inlineCallbacks
+def fetch_deploy_status(config):
+    harold_base_url = config["harold"]["base-url"]
+    if not harold_base_url:
+        returnValue(SAFE_DEFAULT)
+
+    fetch_req = _do_status_http_request(harold_base_url)
```

Comment:
Changing the hostname to a bogus one, this works:

> failed to fetch deploy status: DNS lookup failed: address 'harold.wired.com' not found: [Errno -2] Name or service not known.

The deferreds should be propagating it all, yeah.